internal/querynode/query_node.go

```diff
@@ -135,6 +138,42 @@ func GetQueryNode() *QueryNode {
 	return queryNode
 }
 
+type Hook interface {
+	Run(map[string]interface{}) error
+	Init(string) error
+}
+
+func initHook() Hook {
+	path := Params.QueryNodeCfg.SoPath.GetValue()
+	if path == "" {
+		return nil
+	}
+	log.Debug("start to load plugin", zap.String("path", path))
+	p, err := plugin.Open(path)
+	if err != nil {
+		log.Error("fail to open the plugin, error: ", zap.Error(err))
+		return nil
```

Comment:
if load plugin failed, is it ok to ignore here?

Reply:
yeah, still use the original search parameter provided by user

Reply:
if autoindex enabled, here we need to panic directly